lib/cocoapods/generator/xcconfig/private_pod_xcconfig.rb

```diff
@@ -60,13 +60,13 @@ def generate
             # 'USE_HEADERMAP' => 'NO'
           }
 
-          if target.requires_frameworks?
+          if target.requires_frameworks? && target.scoped?
```

Comment:
Because setting the `CONFIGURATION_BUILD_DIR` here causes that scoped targets are actually scoped.